internal/provider/resource_eva_role.go

```diff
@@ -0,0 +1,270 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/mad-it/terraform-provider-eva/internal/eva"
+)
+
+type roleProviderType struct{}
```

Comment:
Maybe go with something like this since this is not a `provider`

```suggestion
type roleType struct{}
```